src/composables/auth.ts

```diff
@@ -6,97 +6,118 @@ import {
 import { useUi } from './ui';
 import { useModals } from './modals';
+import { useAccounts } from './accounts';
 import { createCustomScopedComposable } from './composablesHelpers';
 
 export const useAuth = createCustomScopedComposable(() => {
   let composableInitialized = false;
-  let isSecureLoginAvailable = false;
+  let isBiometricAuthAvailable = false;
   const isBiometryAvailabilityUpdating = ref(false);
   const isBiometryAvailabilityChecked = ref(false);
   const isAuthenticated = ref(false);
   const isAuthenticating = ref(false);
 
   const {
-    setSecureLoginEnabled,
-    isSecureLoginEnabled,
+    isBiometricLoginEnabled,
     lastTimeAppWasActive,
     secureLoginTimeout,
     isAppActive,
+    setBiometricLoginEnabled,
   } = useUi();
   const { openModal } = useModals();
+  const {
+    isLoggedIn,
+    encryptionKey,
+    setEncryptionKey,
+    openPasswordLoginModal,
+  } = useAccounts();
 
-  async function checkSecureLoginAvailability({ forceUpdate = false } = {}) {
+  /**
+   * Checks if biometric authentication is available on the device.
+   */
+  async function checkBiometricLoginAvailability({ forceUpdate = false } = {}) {
     if (isBiometryAvailabilityUpdating.value) {
       await watchUntilTruthy(() => !isBiometryAvailabilityUpdating.value);
     } else if (!isBiometryAvailabilityChecked.value || forceUpdate) {
       isBiometryAvailabilityUpdating.value = true;
-      isSecureLoginAvailable = (await BiometricAuth.checkBiometry()).isAvailable;
+      isBiometricAuthAvailable = (await BiometricAuth.checkBiometry()).isAvailable;
       isBiometryAvailabilityChecked.value = true;
       isBiometryAvailabilityUpdating.value = false;
     }
-    if (!isSecureLoginAvailable) {
-      setSecureLoginEnabled(false);
+    if (!isBiometricAuthAvailable && isBiometricLoginEnabled.value) {
+      setBiometricLoginEnabled(false);
     }
-    return isSecureLoginAvailable;
+    return isBiometricAuthAvailable;
   }
 
   /**
    * Prompts the user to authenticate using biometric authentication.
    * Returns a promise that resolves when the user is authenticated
    * or if biometric authentication is not available.
    */
-  async function authenticate(): Promise<void> {
-    if (
-      !await checkSecureLoginAvailability()
-      || !isSecureLoginEnabled.value
-      || isAuthenticated.value
-    ) {
+  async function authenticate(password?: string): Promise<void> {
+    if (isAuthenticated.value) {
       return Promise.resolve();
     }
 
-    try {
-      return BiometricAuth.authenticate({
-        reason: t('biometricAuth.reason'),
-        cancelTitle: t('common.cancel'),
-        allowDeviceCredential: true,
-        iosFallbackTitle: t('biometricAuth.fallbackTitle'),
-        androidTitle: t('biometricAuth.title'),
-        androidSubtitle: t('biometricAuth.subtitle'),
-        androidConfirmationRequired: false,
-      }).then(() => {
+    if (IS_MOBILE_APP) {
```

Comment:
I think instead of using one function for both types of authentication it would make sense to have 2 separate functions.
The main reason is that this `authentication` function is used only 2 times in the system: the password login modal and biometric login modal. Both of those modals knows which method they want to use.